network-store-server/src/main/java/com/powsybl/network/store/server/migration/v211limits/V211LimitsMigration.java

```diff
@@ -0,0 +1,270 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.store.server.migration.v211limits;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Stopwatch;
+import com.powsybl.iidm.network.LimitType;
+import com.powsybl.network.store.model.ResourceType;
+import com.powsybl.network.store.model.TemporaryLimitAttributes;
+import com.powsybl.network.store.server.ExtensionHandler;
+import com.powsybl.network.store.server.Mappings;
+import com.powsybl.network.store.server.NetworkStoreRepository;
+import com.powsybl.network.store.server.dto.OwnerInfo;
+import com.powsybl.network.store.server.dto.PermanentLimitAttributes;
+import com.powsybl.network.store.server.exceptions.UncheckedSqlException;
+import liquibase.change.custom.CustomTaskChange;
+import liquibase.database.Database;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.ValidationErrors;
+import liquibase.resource.ResourceAccessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.jdbc.datasource.SingleConnectionDataSource;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static com.powsybl.network.store.server.QueryCatalog.EQUIPMENT_TYPE_COLUMN;
+
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+public class V211LimitsMigration implements CustomTaskChange {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(V211LimitsMigration.class);
+
+    private NetworkStoreRepository repository;
+
+    public void init(Database database) {
+        DataSource dataSource = new SingleConnectionDataSource(((JdbcConnection) database.getConnection()).getUnderlyingConnection(), true);
+        ObjectMapper mapper = new ObjectMapper();
+        this.repository = new NetworkStoreRepository(dataSource, mapper, new Mappings(), new ExtensionHandler(dataSource, mapper));
+    }
+
+    @Override
+    public void execute(Database database) {
+        init(database);
+        JdbcConnection connection = (JdbcConnection) database.getConnection();
+        try (PreparedStatement stmt = connection.prepareStatement("select uuid, variantnum from network ")) {
+            ResultSet variants = stmt.executeQuery();
+            while (variants.next()) {
+                UUID networkUuid = UUID.fromString(variants.getString(1));
+                int variantNum = variants.getInt(2);
+                migrateV211Limits(repository, networkUuid, variantNum);
+            }
+        } catch (SQLException | DatabaseException e) {
+            LOGGER.error(e.getMessage(), e);
```

Comment:
maybe need to be a bit smarter here. We should do the same exception handling as regular sql/xml changesets